torch/_inductor/codegen/triton.py

```diff
@@ -2304,31 +2356,49 @@ def match_block_expr() -> Optional[BlockDescriptorOptions]:
                     if config.triton.use_block_ptr
                     else TensorDescriptorOptions
                 )
-                options = options_class.create(
-                    params=block_params,
-                    constant_offset=offset,
-                    range_trees=range_trees,
-                    mask_vars=mask_vars,
-                    get_max_block=self.max_block,
-                )
 
+                can_lift = False
                 if options_class == TensorDescriptorOptions:
                     nonlocal tma_compatibility_checker
                     tma_compatibility_checker = cast(
                         TMACompatibilityChecker, tma_compatibility_checker
                     )
                     if not tma_compatibility_checker.are_block_parameters_compatible(
-                        options.params
+                        block_params
                     ):
-                        return None
+                        # Some inputs may intend to store the values as (X, Y),
+                        # not (Y, X). As a result, try and tranpsose the blocks
```

Comment:
Do you have any examples of this ? And when they would be tma compatible as Y, X but not X, Y ?